### ISSUE TITLE
An option to use holding down Space key for panning around the canvas

### ISSUE THREAD

**Issue by LxeC** [pencil2d/pencil#1774]
Hello, coming from art programs like Paint Tool SAI, I am so used to **holding down the Space key in order to pan the canvas** around.
Could there be an optional key shortcut added for panning in the Keybindings?

**Comment by MrStevns** [comment]
Hi LxeC

You should already be able to hold space in order to pan the canvas?

**Comment by LxeC** [comment]
> Hi LxeC
> 
> You should already be able to hold space in order to pan the canvas?

Ah I think it didn't work because I had mapped Space to Play/Stop the movie. 
I wonder if it'd be possible to have them both working at the same time? How I would imagine this to work is when I move my cursor over the canvas then pressing and holding space should pan around whereas when I position my cursor over the timeline it should play/stop the animation instead.

**Comment by scribblemaniac** [comment]
That isn't really feasible with the way we currently handle keyboard shortcuts and just overall input focus in the application. Even if it was I don't think that is a direction Pencil2D should go. It would add additional complexity and I could see it quickly becoming another point of confusion for new users asking things along the lines of: "why do certain keyboard shortcuts not work sometimes".

**Comment by LxeC** [comment]
> That isn't really feasible with the way we currently handle keyboard shortcuts and the overall input focus of the application. Even if it was I don't think that is a direction Pencil2D should go. It would add additional complexity and I could see it quickly becoming another point of confusion for new users asking things along the lines of: "why do certain keyboard shortcuts not work sometimes".

I don't believe so, I've seen a few programs that do it similarly with the cursor focus, it works just fine, allows you to do everything with one key easily, and they even have enabled it by default. If the developers are worried about new people confusing things then maybe it just shouldn't be enabled by default, but there's no reason why it couldn't be a user option apart from it hasn't been developed yet.

**Comment by Jose-Moreno** [comment]
@LxeC Hi. First off, thank you for your interest in Pencil2D.

While I won't meddle much into the technical side of the issue, which to my understanding is basically a problem that comes from the base technology used to program the input events, a.k.a Qt Framework and the modifier keys, among other more hardcore things.

To **partially** solve your issue, currently you can map the playback / pause function to CTRL + SHIFT + SPACEBAR. I already tried it and it works great, and it doesn't collide with the other shortcuts. 

FYI a minor reminder:
SPACEBAR -> Pans View
CTRL + SPACEBAR -> Zooms View
ALT + SPACEBAR -> Rotates View

But sure lets say you tie it to `CTRL + SPABEBAR` or even `SPACEBAR` by itself, however doing this will bring a bunch of workflow and user experience issues that I've personally had with professional animation software for the longest time. So I'll present my experience as well as that of my colleagues, hoping it helps provide context.

So, when you are used to using the spacebar to pan the view like myself, you will need to rapidly invoke that functionality because you're constantly moving from one place to the next to draw, that's normal. However playback is different, you would probably need that 1/6th of the time, so it makes more sense to bind it to a less used shortcut in a way that it WON'T trip your workflow when using standard functionality with similar keys. Which in the case we're discussing would be the `SPACEBAR` shortcut which invokes the PAN functionality.

To avoid conflicts and to echo traditional animation methods, we also have **Roll Flipping** & **Inbetween Flipping** functionalities which are meant to be quick behaviors to playback through the specific part you're working in your animation. Try them out you can find them in the Animation Menu, bound to `ALT + X` & `ALT + Z` respectively.

Now I personally have ad the playback mapped to `CTRL + RETURN / ENTER` keys for the longest time because I've worked with Flash for over 20 years, so I have that school of thought & training. But I can see myself binding it to a left handed shortcut so my right hand with the stylus can be freer. How about `SHIFT + SPACEBAR`? just tested that and it does not collide with anything else, and I have been testing it and it doesn't make me trip my fingers when trying to zoom, pan or rotate 👍 

On the other hand to consider your comment on the focus rationale, let me say that I've worked in productions while using Toonboom Harmony as well as Blender, both have a panel focus systems. Blender's is a bit more nuanced, but in Harmony you even have to "enable it" as a sort of magically advanced cantrip.

I have discussed this with colleagues for a long time and having a cursor focus is sometimes a workflow nightmare for the artist, not because it can't be implemented decently but because working with it is annoying when many shortcuts overlap and trying to invoke something will end up doing something else because you forgot you HAD to keep the cursor on a specific panel. It's that simple of a problem. And It becomes a user problem and no matter how much of a gifted memory and spatial awareness one has, everything matters little when you're on a hard due date churning drawings like a madman and you keep tripping on the same problem because the cursor moved slightly once you moved your stylus lol.

Personally I think we should continue upholding the idea seen in programming, and something that I believe we should echo in animation, that one shortcut should have a single responsibility to invoke a specific function. This really helps to develops muscle memory and improves the artist efficiency in return.

Truth be told I'm all for customization and having then user do whatever they want with the software, but binding the same key for everything is not an elegant solution and becomes a problem for both the user and later on for the developers.

That said, I would agree however that hopefully we should be able to find a way to rebind zoom, pan & rotate functionalities to _different_ keys (without the modifiers or the specific mouse buttons), but historically there's been a problem with how the original authors coded these functions so as scribblemaniac I imagine implied before, there would need to be a lot of work involved just to be able to improve on that (though I'm not saying we shouldn't, though we have a bit more critical issues pending to solve 😅 )

I hope my answer is helpful to give perspective, that it somehow helped direct you to a solution for your workflow problem a little bit and that my written tone was friendly enough. I wish you an excellent week.
